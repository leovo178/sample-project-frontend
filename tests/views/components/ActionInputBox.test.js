import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'

import { ActionInputBox } from '@views_components'

describe('ActionInputBox', () => {
	const mockProps = {
		type: 'search',
		placeholder: 'search...',
		defaultValue: '',
		width: '350',
		onSubmit: jest.fn(),
		onChange: jest.fn(),
	}

	afterEach(() => {
		cleanup()
	})

	it('should matche snapshot', () => {
		const props = { ...mockProps }
		const { container } = render(<ActionInputBox {...props} />)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('should render 1 input and 1 button', () => {
		let props = { ...mockProps }
		const { getByPlaceholderText, getByTestId, getByText, rerender } = render(
			<ActionInputBox {...props} />
		)

		expect(getByPlaceholderText(props.placeholder)).toBeTruthy()
		expect(getByTestId('search-icon')).toBeTruthy()

		// change props
		props = { ...props, type: 'create', placeholder: 'text...' }

		rerender(<ActionInputBox {...props} />)
		expect(getByPlaceholderText(props.placeholder)).toBeTruthy()
		expect(getByText('Save')).toBeTruthy()
	})

	it('should call correctly onSubmit function when click button', () => {
		const props = { ...mockProps }
		const { getByTestId } = render(<ActionInputBox {...props} />)

		fireEvent.click(getByTestId('search-icon'))

		expect(props.onSubmit).toHaveBeenCalled()
	})

	it('should change value of input', () => {
		const props = { ...mockProps }
		const { getByPlaceholderText } = render(<ActionInputBox {...props} />)
		const input = getByPlaceholderText(props.placeholder)

		expect(input.value).toBe('')

		fireEvent.change(input, { target: { value: 'Hello world' } })

		expect(input.value).toBe('Hello world')
	})
})
