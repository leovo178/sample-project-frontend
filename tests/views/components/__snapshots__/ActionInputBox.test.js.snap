// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionInputBox should matche snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-7 makeStyles-root-1"
>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-input-2 makeStyles-input-5"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        placeholder="search..."
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-46 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-47"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-containedSizeLarge MuiButton-sizeLarge"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-4"
        data-testid="search-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
